apps/examples/timer/timer_main.c

```diff
@@ -106,11 +112,44 @@ static pthread_addr_t timer_thread(pthread_addr_t arg)
 
 	pthread_sigmask(SIG_BLOCK, &sig_set, NULL);
 
+	/*
+	 * Register a callback for notifications using the configured signal.
+	 * NOTE: If no callback is attached, the timer stop at the first interrupt.
+	 */
+	fprintf(stdout, "Attach timer handler\n");
+
+	notify.arg   = NULL;
+	notify.pid   = (pid_t)getpid();
+	notify.signo = EXAMPLE_TIMER_SIGNO;
+
+	if (ioctl(fd, TCIOC_NOTIFICATION,
+			(unsigned long)((uintptr_t)&notify)) < 0) {
+		fprintf(stderr, "ERROR: Failed to set the timer handler: %d\n", errno);
+		goto error;
+	}
+
+	/* Start the timer */
+	fprintf(stdout, "Start the timer\n");
+	if (ioctl(fd, TCIOC_START, 0) < 0) {
+		fprintf(stderr, "ERROR: Failed to start the timer: %d\n", errno);
+		goto error;
+	}
+
```

Comment:
There is a possible race between START and SETTIMEOUT since SETTIMEOUT is done in main task after thread creation. How about moving SETTIMEOUT to this thread as well?

Reply:
Yes, i agree